File: app/src/main/java/ru/iandreyshev/parserrss/models/repository/Article.java
package ru.iandreyshev.parserrss.models.repository;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import java.util.Date;

import io.objectbox.annotation.Convert;
import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;

@Entity
public final class Article {
    @Id
    long mId;
    long mRssId;

    String mTitle = "";
    String mOriginUrl = "";
    String mDescription = "";
    String mImageUrl;
    Long mPostDate;
    @Convert(dbType = byte[].class, converter = BitmapConverter.class)
    Bitmap mImage;

    public Article(@NonNull String title, @NonNull String description, @NonNull String originUrl) {
        mTitle = title;
        mDescription = description;
        mOriginUrl = originUrl;
    }

    Article() {
    }

    public long getId() {
        return mId;
    }

    @NonNull
    public String getTitle() {
        return mTitle;
    }

    @NonNull
    public String getOriginUrl() {
        return mOriginUrl;
    }

    @NonNull
    public String getDescription() {
        return mDescription;
    }

    public Long getDate() {
        return mPostDate;
    }

    public String getImageUrl() {
        return mImageUrl;
    }

    public Bitmap getImage() {
        return mImage;
    }

    public Long getRssId() {
        return mRssId;
    }

    void setRssId(long id) {
        mRssId = id;
    }

    public void setImage(final Bitmap image) {
        mImage = image;
    }

    public void setDate(final Date date) {
        mPostDate = date.getTime();
    }

    public void setImageUrl(final String url) {
        mImageUrl = url;
    }

    @Override
    public final boolean equals(Object other) {
        return (other instanceof Article) && mOriginUrl.equals(((Article) other).mOriginUrl);
    }

    @Override
    public final int hashCode() {
        return mOriginUrl.hashCode();
    }
}

File: app/src/main/kotlin/ru/iandreyshev/parserrss/models/repository/extention/Article.kt
package ru.iandreyshev.parserrss.models.repository.extention

import ru.iandreyshev.parserrss.models.repository.ArticleEntity
import ru.iandreyshev.parserrss.models.rss.Article

internal val ArticleEntity.domainModel
    get() = Article(
            id = this.id,
            rssId = this.rssId,
            title = this.title,
            description = this.description,
            date = this.date,
            originUrl = this.originUrl,
            imageUrl = this.imageUrl
    )

internal val Article.entity
    get() = ArticleEntity(
            id = this.id,
            title = this.title,
            description = this.description,
            date = this.date,
            originUrl = this.originUrl,
            imageUrl = this.imageUrl
    )
